src/dx/pubnub_client.rs

```diff
@@ -227,9 +227,8 @@ where
                 Ok(PubNubClientRef {
                     transport: PubNubMiddleware {
                         transport: pre_build.transport,
-                        // TODO: String -> Cow<'static, str>
-                        instance_id: pre_build.instance_id.clone(),
-                        user_id: pre_build.config.user_id.clone(),
+                        instance_id: Arc::new(pre_build.instance_id.clone()),
```

Comment:
@Xavrax  I think compiler doesn't like this way of increasing reference count for Option string type

Reply:
wrapping in `Some` looks relevant but still it was mismatch of type.

Reply:
Can I ask why `Arc`?

Reply:
this field in middleware should get reference (not clone)
since this instance is getting created and returned by build method. If we borrow it here, it's against rust's rules - it says that we can not returned value partially owned(`instance_id` field) by this function.

So reference counted string seemed proper choice here. And in async environment we can not use `Rc` so that's why it's `Arc`

Mateusz can correct me if it's not correct reason.;-)

Reply:
It is probably one of the use cases. I know only that `Arc` is good for situations, where shared resources (allocated on heap) used from multiple places and should be valid as long as it in use (even after leaving scope, where it has been allocated).

Reply:
Yes, we want to share this property value between middleware and client. at least that was intention

Reply:
`Arc` mostly provide protection against deallocation. I would say, that in the current situation, as long as the instance from which we borrow this `String` is “alive”, there is no need for resource deallocation protection. Also `Arc` adds some performance overhead compared to regular memory access.

Reply:
agree!
the problem what I experienced in this scenario  was that we are returning value which is referencing data in builder!! So when I had defined `'client` lifetime. It was still not able to guarantee it.